feat(core): Add log streaming events for executions

MarcL opened the pull request:
## Summary

Adds two new log streaming events for audit logging:
- `n8n.audit.user.execution.deleted`: Emitted when user deletes executions
- `n8n.audit.user.workflow.executed`: Emitted when user manually runs or retries a workflow

Both events include user context for audit tracking.

## Related Linear tickets, Github issues, and Community forum posts

https://linear.app/n8n/issue/PAY-2936


## Review / Merge checklist

- [x] PR title and summary are descriptive. ([conventions](../blob/master/.github/pull_request_title_conventions.md)) <!--
   **Remember, the title automatically goes into the changelog.
   Use `(no-changelog)` otherwise.**
-->
- [ ] [Docs updated](https://github.com/n8n-io/n8n-docs) or follow-up ticket created.
- [x] Tests included. <!--
   A bug is not considered fixed, unless a test is added to prevent it from happening again.
   A feature is not complete without tests.
-->
- [ ] PR Labeled with `release/backport` (if the PR is an urgent fix that needs to be backported)

MarcL committed:
feat(core): Add log streaming events for execution deletion and workflow execution (PAY-2936)

- Add 'execution-deleted' event when user deletes executions
- Add 'workflow-executed' event when user manually runs or retries workflow
- Emit events with user context for audit logging
- Add comprehensive unit tests for both events

MarcL committed:
fix: Include deleteBefore in execution-deleted event payload for complete audit trail

When executions are deleted via date filter (deleteBefore), the executionIds
array would be empty. Including deleteBefore in the event payload provides
a complete audit trail showing what criteria was used for bulk deletion.

MarcL committed:
refactor: Move workflow-executed event to Workflow region in relay.event-map.ts

MarcL committed:
refactor(core): Omit deleteBefore from execution-deleted event when not set

Only include deleteBefore in the payload when it has a value,
rather than including deleteBefore: undefined.

MarcL committed:
refactor(core): Move workflow-executed handler to Workflow region

Organize the workflow-executed event listener and handler alongside
other workflow events for better code organization.

MarcL committed:
refactor(core): Rename workflow-executed to workflow-executed-by-user

Make the relay event name more explicit that this event is specifically
for user-triggered executions (manual runs and retries), not for all
workflow executions like scheduled or webhook-triggered workflows.

This addresses PR feedback about clarity in the relay event map.

MarcL commented on packages/cli/src/events/maps/relay.event-map.ts:
```diff
@@ -125,6 +125,14 @@ export type RelayEventMap = {
+		workflowId: string;
+		workflowName: string;
+		executionId: string;
+		executionMode: 'manual' | 'retry';
```
@BGZStephen How about I refactor this to go back to the original workflow executed event and add the source? I might just scope this PR to the user-related changes and then add a 2nd PR to do the non-user events. Thoughts?

And should we do the source like:

`'user' | 'schedule' | 'webhook' | 'error' | 'trigger'`

And then keep the execution mode? `'manual' | 'retry'`

Or use a specific source for manual or retry and no execution mode?
E.g. `'user-manual' | 'user-retry' | 'schedule' | 'webhook' | 'error' | 'trigger'`

MarcL committed:
refactor(core): Use generic workflow-executed event with source field

Replace workflow-executed-by-user with a generic workflow-executed event
that includes a source field to distinguish execution triggers. This
enables future expansion to non-user sources (webhook, trigger, etc.)
while keeping the current scope to user-triggered executions.

Changes:
- Rename event to workflow-executed with optional user field
- Add source field with all possible execution sources
- Update audit event name to n8n.audit.workflow.executed
- Current sources: user-manual, user-retry (others for future use)

konstantintieber commented on packages/cli/src/events/maps/relay.event-map.ts:
```diff
@@ -125,12 +125,22 @@ export type RelayEventMap = {
-		executionMode: 'manual' | 'retry';
+		source:
+			| 'user-manual'
+			| 'user-retry'
```
I understand that this typing includes values for `source` that are not implemented yet, right?
Is it planned do still add these imminently in a follow-up?
Otherwise keeping them here might lead to confusion on consumers of log streams seeing these values here, but never in the logs.

An alternative I see would be to leave them commented out, while not implemented.

MarcL replied:
Good point @konstantintieber.
My plan was to quickly do a 2nd PR afterwards. I was trying to avoid making this PR too big.
I'll comment them out in this PR and add them back in the next.

MarcL committed:
refactor(core): Comment out future source values in workflow-executed event

Keep only user-manual and user-retry as active sources for now,
with other sources commented out for future PR implementation.
